Build failure looks real:
```
ERROR: corehq.sql_db.tests.test_connections:ConnectionManagerTests.test_filter_out_stale_standbys
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/sql_db/tests/test_connections.py", line 153, in test_filter_out_stale_standbys
    filter_out_stale_standbys(['ucr', 'default']),
  File "/vendor/lib/python2.7/site-packages/quickcache/quickcache.py", line 42, in inner
    return helper(*args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/quickcache/quickcache_helper.py", line 181, in __call__
    content = self.fn(*args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/sql_db/util.py", line 272, in filter_out_stale_standbys
    if get_replication_delay_for_standby(db, relevant_dbs) <= delays_by_db.get(db, ACCEPTABLE_STANDBY_DELAY_SECONDS)
TypeError: <lambda>() takes exactly 1 argument (2 given)
```

--- a/corehq/sql_db/tests/test_connections.py
+++ b/corehq/sql_db/tests/test_connections.py
@@ -148,7 +148,7 @@ class ConnectionManagerTests(SimpleTestCase):
         )
 
     def test_filter_out_stale_standbys(self, *args):
-        with mock.patch('corehq.sql_db.util.get_replication_delay_for_standby', lambda x: {'ucr': 2, 'default': 4}.get(x, 0)):
+        with mock.patch('corehq.sql_db.util.get_replication_delay_for_standby', lambda x, y: {'ucr': 2, 'default': 4}.get(x, 0)):
             self.assertEqual(
                 filter_out_stale_standbys(['ucr', 'default']),
                 ['ucr']

--- a/corehq/sql_db/tests/test_connections.py
+++ b/corehq/sql_db/tests/test_connections.py
@@ -99,7 +99,7 @@ class ConnectionManagerTests(SimpleTestCase):
                 [manager.get_load_balanced_read_db_alias('default') for i in range(3)]
             )
 
-    @mock.patch('corehq.sql_db.util.get_replication_delay_for_standby', lambda x: {'ucr': 4}.get(x, 0))
+    @mock.patch('corehq.sql_db.util.get_replication_delay_for_standby', lambda x, y: {'ucr': 4}.get(x, 0))
     def test_standby_filtering(self, *args):
         reporting_dbs = {
             'ucr_engine': {

--- a/corehq/sql_db/util.py
+++ b/corehq/sql_db/util.py
@@ -282,7 +282,7 @@ def filter_out_stale_standbys(dbs):
     # from given list of databases filters out those with more than
     #   acceptable standby delay, if that database is a standby
     delays_by_db = get_standby_delays_by_db()
-    relevant_dbs = tuple([pair[0] for pair in dbs])
+    relevant_dbs = tuple(dbs)
     return [
         db
         for db in dbs
@@ -308,7 +308,7 @@ def select_db_for_read(weighted_dbs):
     weights_by_db = {_db: weight for _db, weight in weighted_dbs}
 
     # filter out stale standby dbs
-    fresh_dbs = filter_out_stale_standbys(weights_by_db)
+    fresh_dbs = filter_out_stale_standbys(list(weights_by_db.keys()))
     dbs = []
     weights = []
     for _db, weight in six.iteritems(weights_by_db):

